keychain_test.go

```diff
@@ -103,7 +105,26 @@ func TestOSXKeychainKeyringOverwrite(t *testing.T) {
 	}
 }
 
-func TestOSXKeychainKeyringListKeys(t *testing.T) {
+func TestOSXKeychainKeyringListKeysWhenEmpty(t *testing.T) {
+	path := tempPath()
+	defer deleteKeychain(path, t)
+
+	k := &keychain{
+		path:         path,
+		service:      "test",
+		passwordFunc: fixedStringPrompt("test password"),
+		isTrusted:    true,
+	}
+
+	_, err := k.Keys()
+	// TODO: we should consider making a generic package error like keyring.ErrNoKeyChain instead of letting
```

Comment:
Yeah, let's do that

Reply:
Perhaps with wrapping from https://godoc.org/github.com/pkg/errors? I guess type checking gets harder then. We could go for `keychain.IsErrNoKeyChain()`?

Reply:
Will do. I had a thought that perhaps something along the lines of `BackendNotInitialized` instead of 'NoKeyChain' since "not initialized" is more generic.

RE: pkg/errors. I don't follow. Can you provide an example? Using the existing pattern for no such key is easy to check for errors:

```
i, err := s.kr.Get(vaultAddr)
		if err == keyring.ErrKeyNotFound {
			...
		}
```

Reply:
Will do. I had a thought that perhaps something along the lines of `BackendNotInitialized` instead of 'NoKeyChain' since "not initialized" is more generic.

RE: pkg/errors. I don't follow. Can you provide an example? Using the existing pattern for no such key is easy to check for errors:

```
i, err := kr.Get(vaultAddr)
		if err == keyring.ErrKeyNotFound {
			...
		}
```

Reply:
All good, ignore me for now, I think constant errors are fine here.